File: src/main/java/com/projeto/springapi/model/FotoPessoa.java
package com.projeto.springapi.model;

import jakarta.persistence.*;
import java.time.LocalDate;
import lombok.Data;

@Entity
@Table(name = "foto_pessoa")
@Data
public class FotoPessoa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "fp_id")
    private Long fpId;

    @Column(name = "fp_data")
    private LocalDate fpData;

    @Column(name = "fp_bucket")
    private String fpBucket;

    @Column(name = "fp_hash")
    private String fpHash;

    @OneToOne
    @JoinColumn(name = "pes_id", unique = true)
    private Pessoa pessoa;

}

File: app/Models/FotoPessoa.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class FotoPessoa extends Model
{
    protected $table = 'foto_pessoa';
    protected $primaryKey = 'fp_id';
    public $timestamps = false;

    protected $fillable = [
        'fp_data',
        'fp_bucket',
        'fp_hash',
        'pes_id'
    ];

    public function pessoa()
    {
        return $this->belongsTo(Pessoa::class, 'pes_id', 'pes_id');
    }
}
